# https://bugzilla.mozilla.org/show_bug.cgi?id=1270036
From bb82fd3fc1f2f7d4beb8ea9a6cf34f272027bf96 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 4 May 2016 13:41:09 +0200
Subject: [PATCH] Fix the zlib module in SM tarballs


diff --git a/js/src/make-source-package.sh b/js/src/make-source-package.sh
index ab53a85..e7e2caa 100755
--- a/js/src/make-source-package.sh
+++ b/js/src/make-source-package.sh
@@ -105,8 +105,8 @@ case $cmd in
 		${tgtpath}/testing
 	${MKDIR} -p ${tgtpath}/modules
 	cp -pPR \
-		${TOPSRCDIR}/modules/zlib/src \
-		${tgtpath}/modules
+		${TOPSRCDIR}/modules/zlib/src/ \
+		${tgtpath}/modules/zlib
 	${MKDIR} -p ${tgtpath}/layout/tools/reftest
 	cp -pPR \
 		${TOPSRCDIR}/layout/tools/reftest/reftest \
-- 
2.8.1
