commit cc459eeccebb6289da3ae8b34639fa429011e377
Author: gnaggnoyil <gnaggnoyil@gmail.com>
Date:   Sun Feb 25 22:06:23 2018 +0800

    Enable detection for libc++7.0's deprecated <experimental/string_view>

diff --git a/include/boost/asio/detail/config.hpp b/include/boost/asio/detail/config.hpp
index 126be7e6..518f501c 100644
--- a/include/boost/asio/detail/config.hpp
+++ b/include/boost/asio/detail/config.hpp
@@ -773,7 +773,9 @@
 # endif // !defined(BOOST_ASIO_DISABLE_STD_FUTURE)
 #endif // !defined(BOOST_ASIO_HAS_STD_FUTURE)
 
-// Standard library support for std::string_view.
+// Standard library support for experimental::string_view.
+// Note: libc++ 7.0 keeps <experimental/string_view> header but deprecates it
+// with an #error directive.
 #if !defined(BOOST_ASIO_HAS_STD_STRING_VIEW)
 # if !defined(BOOST_ASIO_DISABLE_STD_STRING_VIEW)
 #  if defined(__clang__)
@@ -804,8 +806,14 @@
 #  if defined(__clang__)
 #   if (__cplusplus >= 201402)
 #    if __has_include(<experimental/string_view>)
-#     define BOOST_ASIO_HAS_STD_EXPERIMENTAL_STRING_VIEW 1
+#     if defined(_LIBCPP_LFTS_STRING_VIEW)
+#      define BOOST_ASIO_HAS_STD_EXPERIMENTAL_STRING_VIEW 1
+#      define BOOST_ASIO_HAS_STD_STRING_VIEW 1
+#     endif // defined(_LIBCPP_LFTS_STRING_VIEW)
 #    endif // __has_include(<experimental/string_view>)
+#    if __has_include(<string_view>)
+#     define BOOST_ASIO_HAS_STD_STRING_VIEW 1
+#    endif // __has_include(<string_view>)
 #   endif // (__cplusplus >= 201402)
 #  endif // defined(__clang__)
 #  if defined(__GNUC__)
